From 15bf414daa83967fd1f24bd48bd01ea941c11ce2 Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter.hutterer@redhat.com>
Date: Thu, 13 Nov 2008 10:17:33 +1000
Subject: [PATCH] xfree86: AllowEmptyInput is true by default - update the xf86Info defaults.

Also set AutoAddDevices and AutoEnableDevices to their defaults.

And in doing so, switch the rest of the defaults over to named intializers.

Signed-off-by: Peter Hutterer <peter.hutterer@redhat.com>
(cherry picked from commit 0b56b44addc323a00eb7cd86240cb0dd4275bcf8)

Conflicts:

	hw/xfree86/common/xf86Globals.c
---
 hw/xfree86/common/xf86Globals.c |   69 ++++++++++++++++++++------------------
 1 files changed, 36 insertions(+), 33 deletions(-)

diff --git a/hw/xfree86/common/xf86Globals.c b/hw/xfree86/common/xf86Globals.c
index f5babbc..a5edd82 100644
--- a/hw/xfree86/common/xf86Globals.c
+++ b/hw/xfree86/common/xf86Globals.c
@@ -95,45 +95,48 @@ InputInfoPtr xf86InputDevs = NULL;
 /* Globals that video drivers may not access */
 
 xf86InfoRec xf86Info = {
-	-1,		/* consoleFd */
-	-1,		/* vtno */
-	FALSE,		/* vtSysreq */
-	SKWhenNeeded,	/* ddxSpecialKeys */
-	NULL,		/* pMouse */
+    .consoleFd                  = -1,
+    .vtno                       = -1,
+    .vtSysreq                   = FALSE,
+    .ddxSpecialKeys             = SKWhenNeeded,
+    .pMouse                     = NULL,
 #ifdef XINPUT
-	NULL,		/* mouseLocal */
+    .mouseLocal                 = NULL,
 #endif
-	-1,		/* lastEventTime */
-	FALSE,		/* vtRequestsPending */
-	FALSE,		/* inputPending */
-	FALSE,		/* dontVTSwitch */
-	FALSE,		/* dontZap */
-	FALSE,		/* dontZoom */
-	FALSE,		/* notrapSignals */
-	FALSE,		/* caughtSignal */
-	FALSE,		/* sharedMonitor */
-	NULL,		/* currentScreen */
+    .lastEventTime              = -1,
+    .vtRequestsPending          = FALSE,
+    .inputPending               = FALSE,
+    .dontVTSwitch               = FALSE,
+    .dontZap                    = FALSE,
+    .dontZoom                   = FALSE,
+    .notrapSignals              = FALSE,
+    .caughtSignal               = FALSE,
+    .sharedMonitor              = FALSE,
+    .currentScreen              = NULL,
 #ifdef CSRG_BASED
-	-1,		/* screenFd */
-	-1,		/* consType */
+    .screenFd                   = -1,
+    .consType                   = -1,
 #endif
-	FALSE,		/* allowMouseOpenFail */
-	TRUE,		/* vidModeEnabled */
-	FALSE,		/* vidModeAllowNonLocal */
-	TRUE,		/* miscModInDevEnabled */
-	FALSE,		/* miscModInDevAllowNonLocal */
-	PCIOsConfig,	/* pciFlags */
-	Pix24DontCare,	/* pixmap24 */
-	X_DEFAULT,	/* pix24From */
+    .allowMouseOpenFail         = FALSE,
+    .vidModeEnabled             = TRUE,
+    .vidModeAllowNonLocal       = FALSE,
+    .miscModInDevEnabled        = TRUE,
+    .miscModInDevAllowNonLocal  = FALSE,
+    .pciFlags                   = PCIOsConfig,
+    .pixmap24                   = Pix24DontCare,
+    .pix24From                  = X_DEFAULT,
 #ifdef __i386__
-	FALSE,		/* pc98 */
+    .pc98                       = FALSE,
 #endif
-	TRUE,		/* pmFlag */
-	LogNone,	/* syncLog */
-	0,		/* estimateSizesAggressively */
-	FALSE,		/* kbdCustomKeycodes */
-	FALSE,		/* disableRandR */
-	X_DEFAULT	/* randRFrom */
+    .pmFlag                     = TRUE,
+    .log                        = LogNone,
+    .estimateSizesAggressively  = 0,
+    .kbdCustomKeycodes          = FALSE,
+    .disableRandR               = FALSE,
+    .randRFrom                  = X_DEFAULT,
+    .allowEmptyInput            = TRUE,
+    .autoAddDevices             = TRUE,
+    .autoEnableDevices          = TRUE
 };
 const char *xf86ConfigFile = NULL;
 const char *xf86InputDeviceList = NULL;
-- 
1.6.0.3

